segmentation/include/pcl/segmentation/extract_clusters.h

```diff
@@ -325,7 +335,7 @@ namespace pcl
       EuclideanClusterExtraction () : tree_ (), 
                                       cluster_tolerance_ (0),
                                       min_pts_per_cluster_ (1), 
-                                      max_pts_per_cluster_ (std::numeric_limits<int>::max ())
+                                      max_pts_per_cluster_ (std::numeric_limits<unsigned int>::max ())
```

Comment:
Maybe `size_t` or `uindex_t` would be better for the sizes in this file?

Reply:
Because of the extended range? Personally I think `unsigned int` should be large enough, but I would also be ok with `size_t` or `uindex_t`.

Reply:
Maybe we should go for either of the suggested, as unsigned int on windows is still only 4 bytes.
See https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=msvc-160.

This seems also the case for linux?

Since max size is the limit of the index's I would vote for index_t, as it then follows the index size.